clients/unshaded/src/main/java/tachyon/client/file/FileOutStream.java

```diff
@@ -79,12 +80,20 @@
    * @throws IOException if an I/O error occurs
    */
   public FileOutStream(long fileId, OutStreamOptions options) throws IOException {
+    this(fileId, options, FileSystemContext.INSTANCE);
+  }
+
+  // Test-only constructor for mocking the file system context. Whitebox cannot be used here because
+  // the context is used before the constructor finishes.
+  FileOutStream(long fileId, OutStreamOptions options, FileSystemContext context)
```

Comment:
You should still be able to get this to work with powermock by mocking the FileSystemContext instance?